include/zephyr/drivers/seu.h

```diff
@@ -0,0 +1,442 @@
+/**
+ * Copyright (c) 2023 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Public Single Event Upsets(SEU) driver APIs
+ */
+
+#ifndef ZEPHYR_INCLUDE_SEU_H_
+#define ZEPHYR_INCLUDE_SEU_H_
+
+/**
+ * @brief Single Event Upsets(SEU) Interface
+ * @defgroup single_event_upsets_interface Single Event Upsets(SEU) Interface
+ * @ingroup io_interfaces
+ * @{
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define SEU_MAX_FRAME 2
+
+/**
+ * Structure to hold user input to inject safe SEU error frame data.
+ */
+struct inject_safe_seu_error_frame {
```

Comment:
Correct me if I'm wrong, but these frame structs seem to be tailored to one specific driver. While I understand that the driver you're adding in this PR is the only one we have so far, could we make this API more generic?